pontoon/base/tests/test_models.py

```diff
@@ -0,0 +1,167 @@
+from datetime import datetime
+
+from django_nose.tools import assert_equal, assert_true
+from django.test.utils import override_settings
+
+from pontoon.base.models import User
+from pontoon.base.tests import (
+    IdenticalTranslationFactory,
+    TranslationFactory,
+    TestCase,
+    UserFactory)
+
+
+class UserTranslationManagerTests(TestCase):
+    @override_settings(EXCLUDE=('excluded@example.com',))
+    def test_excluded_contributors(self):
+        """
+        Checks if contributors with mails in settings.EXCLUDE are excluded
+        from top contributors list.
+        """
+        included_contributor = TranslationFactory.create(user__email='included@example.com').user
+        excluded_contributor = TranslationFactory.create(user__email='excluded@example.com').user
+
+        top_contributors = User.translators.with_translation_counts()
+        assert_true(included_contributor in top_contributors)
+        assert_true(excluded_contributor not in top_contributors)
+
+    def test_users_without_translations(self):
+        """
+        Checks if user contributors without translations are returned.
```

Comment:
Yup, you're right.

On Thu, Sep 17, 2015 at 11:42 PM, Michael Kelly <notifications@github.com>
wrote:

> In pontoon/base/tests/test_models.py
> <https://github.com/mozilla/pontoon/pull/161#discussion_r39802822>:
>
> > +    @override_settings(EXCLUDE=('excluded@example.com',))
> > +    def test_excluded_contributors(self):
> > +        """
> > +        Checks if contributors with mails in settings.EXCLUDE are excluded
> > +        from top contributors list.
> > +        """
> > +        included_contributor = TranslationFactory.create(user__email='included@example.com').user
> > +        excluded_contributor = TranslationFactory.create(user__email='excluded@example.com').user
> > +
> > +        top_contributors = User.translators.with_translation_counts()
> > +        assert_true(included_contributor in top_contributors)
> > +        assert_true(excluded_contributor not in top_contributors)
> > +
> > +    def test_users_without_translations(self):
> > +        """
> > +        Checks if user contributors without translations are returned.
>
> I think you mean they *aren't* included.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mozilla/pontoon/pull/161/files#r39802822>.
>



-- 
jarek@reijutsu:~$ fortune
You have Egyptian flu: you're going to be a mummy.
jarek@reijutsu:~$ fortune
You now have Asian Flu.